### DISCUSSION TITLE
Measurement jump (by 1m³) due to early last digit increment

### DISCUSSION THREAD

**Discussion by feci2048** [jomjol/AI-on-the-edge-device#3433] [General]
### The Problem

I am on 16.0.0-RC5, but only upgraded to it during the issue.
The misbehaviour started with 16.0.0-RC4.

I'm not sure if my case qualifies as an early transition, because with the analog hands showing (correclty) 0.9xxx, it's to be expected that the last digit already looks like a "perfect" 0, even if it still could rotate 1 pixel row to become "perfect".  
The max flow rate was at 40 l/minute, which somehow allowed for this transition to occur, now I further reduced it to 35 l/minute. Although theoretically I could get to 38 l/minute in the household+garden, so I wont be able to use it as a workaround.

I haven't noticed this behaviour until today, been running the setup for around 10 months now.

I was hoping that the expert parameter "Check Digit Increase Consistency" (which is active) tries to fix exactly these sort of issues.

I'm not sure if a fix can be implemented, but if the last digit reads as 0.0 (and definetly < 0.5) and at the same time the analog reading is 0.9+, then it's safe to assume the last digit is still 9. 

[data.zip](https://github.com/user-attachments/files/18079194/data.zip)

If I can help with any additional data, let me know.

### Version

v16.0.0-RC5 (Commit: 7836323+)

### Logfile

```shell
Logfile is gone, because I upgraded during the issue to v16.0.0-RC5 from v16.0.0-RC4, but context is provided in the attached data.zip
```

### Expected Behavior

If last Digit is recognised as less than 0.5 (even 0.0), while Analog counter indicate 0.9+, then interpret Digit 0 as 9 instead.
The jump in Value of 1 m³ suddenly, should be filtered out by the max flowrate limit.

### Screenshots

Included in attached zip file.

**Comment by caco3** [comment]
It would be helpful if you would provide a screenshot and the configuration.

In general, [AnalogToDigitTransitionStart](https://jomjol.github.io/AI-on-the-edge-device-docs/Parameters/#parameter-numberanalogtodigittransitionstart) is exactlty what should prevent this issue.

What was the version last version on which you did **not** see this issue? Can you go back to that version and check if it is back ok?

I doubt it is regression of the software but a configuration issue. Some meters are really to handle properly. Some even are impossible. But as you say it worked for a long time, maybe just the config needs some tweaking.

**Comment by feci2048** [comment]
In the attached data.zip file I've included config.ini, measurement csv, pictures, screenshot.
I was using 15.7.0 since February (I set up the whole system directly with this version).
After going to 16.0.0-RC4 I did notice changes, like much brighter pictures (I attributed that to auto-gain), and other image processing changes, but I reconfigured and moved on.
Then the issue happened, and while the issue was persisting I did the upgrade to RC5, just to see if the handling there is different, but it turned out to be irrecoverable.
Of course once the analog meters overflowed from 0.9xxx to 1(0), the Digital remained 0, so the jump "corrected" itself.
If you could look at data.zip attached to the first post. 
All data is logged into Homeassistant and I look often (push notifications) if something is wrong with the watermeter and I didn't see this sort of issue ever. There were other issues, but rather because my own less than ideal calibration and lighting.

EDIT: You are correct, according to my own config.ini the "Analog to Digit Transition Start" is unconfigured, but it quite possible that it could've saved this situation if configured to 9.9. I will configure it as such and look what happens next time.

**Reply by caco3** [reply]
I moved it to a discussion because I think it is just a config issue.
Feel free to respond when you have new information.
